docs: Add up-to-date documentation and remove legacy documentation (DEV-4229) (#263)
dasch-swiss/dsp-meta: docs/data/datamodel.md

| `__type`          | string      | 1           | Literal 'Organization'                 |
| `name`            | string      | 1           |                                        |
| `url`             | url         | 1           |                                        |
| `address`         | address     | 0-1         |                                        |
| `email`           | string      | 0-1         |                                        |
| `alternativeName` | lang_string | 0-1         |                                        |
| `authorityRefs`   | url[]       | 0-n         | References to external authority files |

#### Grant

| Field     | Type   | Cardinality | Restrictions               |
| --------- | ------ | ----------- | -------------------------- |
| `__id`    | string | 1           |                            |
| `__type`  | string | 1           | Literal 'Grant'            |
| `funders` | id[]   | 1-n         | Person or Organization IDs |
| `number`  | string | 0-1         |                            |
| `name`    | string | 0-1         |                            |
| `url`     | url    | 0-1         |                            |

### Value Types

> #### lang_string

Reviewer: ```suggestion
#### Language Tag (`lang_string')
```

Author: Thanks. I'm going for `#### String with Language Tag (lang_string)` instead though, to make it clear that it's not _only_ the tag, but the string with the tag.

> Object with a language tag as key and a string as value.

Reviewer 2: ```suggestion
Object with a language ISO code as key and a string as value.
```

```json
{
    "en": "Lorem ipsum in English.",
    "de": "Lorem ipsum auf Deutsch."
}
```

> #### date

Reviewer: ```suggestion
#### Date
```

String with the format `YYYY-MM-DD`.

> #### url

Reviewer: ```suggestion
#### URL
```

An object representing a URL. 
Depending on the `type` field,
the URL may be a generic URL

> or a more specific link, like P PID

Reviewer 2: ```suggestion
or a more specific link, like PID
```

Author: meant to write "like a PID", thanks for spotting.

or a reference to a resource in an external authority file.

| Field    | Type   | Cardinality | Restrictions                                                                                                                                |
| -------- | ------ | ----------- | ------------------------------------------------------------------------------------------------------------------------------------------- |
| `__type` | string | 1           | Literal 'URL'                                                                                                                               |
| `type`   | string | 1           | Literal 'URL', 'Geonames', 'Pleiades', 'Skos', 'Periodo', 'Chronontology', 'GND', 'VIAF', 'Grid', 'ORCID', 'Creative Commons', 'DOI', 'ARK' |
| `url`    | string | 1           |                                                                                                                                             |
| `text`   | string | 0-1         |                                                                                                                                             |

> #### dmp (Data Management Plan)

Reviewer: ```suggestion
#### Data Management Plan (`dmp`)
```

| Field      | Type    | Cardinality | Restrictions                 |
| ---------- | ------- | ----------- | ---------------------------- |
| `__type`   | string  | 1           | Literal 'DataManagementPlan' |
| `available | boolean | 0-1         |                              |
| `url`      | url     | 0-1         |                              |

> #### publication

Reviewer: ```suggestion
#### Publication
```

| Field  | Type   | Cardinality | Restrictions |
| ------ | ------ | ----------- | ------------ |
| `text` | string | 1           |              |
| `url`  | url    | 0-1         |              |

> #### address

Reviewer: ```suggestion
#### Address
```

| Field        | Type   | Cardinality | Restrictions      |
| ------------ | ------ | ----------- | ----------------- |
| `__type`     | string | 1           | Literal 'Address' |
| `street`     | string | 1           |                   |
| `postalCode` | string | 1           |                   |
| `locality`   | string | 1           |                   |
| `country`    | string | 1           |                   |
| `canton`     | string | 0-1         |                   |
| `additional` | string | 0-1         |                   |

> #### license

Reviewer: ```suggestion
#### License
```

| Field     | Type   | Cardinality | Restrictions      |
| --------- | ------ | ----------- | ----------------- |
| `__type`  | string | 1           | Literal 'License' |
| `license` | url    | 1           |                   |
| `date`    | date   | 1           |                   |
| `details` | string | 0-1         |                   |

> #### attribution

Reviewer: ```suggestion
#### Attribution
```

        lang_string_or_url[] additional "0-n"
    }

    person {
        string __id "1"
        string __type "1; Literal 'Person'"
        string[] givenNames "1-n"
        string[] familyNames "1-n"
        string[] jobTitles "0-n"
        id[] affiliations "0-n; Organization IDs"
        address address "0-1"
        string email "0-1"
        string secondaryEmail "0-1"
        url[] authorityRefs "0-n"
    }

    organization {
        string __id "1"
        string __type "1; Literal 'Organization'"
        string name "1"
        url url "1"
        address address "0-1"
        string email "0-1"
        lang_string alternativeName "0-1"
        url[] authorityRefs "0-n"
    }

    grant {
        string __id "1"
        string __type "1; Literal 'Grant'"
        id[] funders "1-n; Person or Organization IDs"
        string number "0-1"
        string name "0-1"
        url url "0-1"
    }
```

## Legacy docs

The legacy documentation can be found [here](https://github.com/dasch-swiss/dsp-meta-svc/tree/main/docs/services/metadata).
This includes thoughts on mappings to external aggregators 
and on RDF representation of the metadata.

## Issues/Questions

> - access conditions on datasets not current state
> - status on datasets not aligned with project status
> - typeOfData on datasets is a weird list of options
> - does dataset need all the date fields?

Reviewer 2: ```suggestion
- access conditions on datasets not current state,
- status on datasets not aligned with project status,
- typeOfData on datasets is a weird list of options,
- does dataset need all the date fields?
```

Author: I actually meant to remove this entirely, before merging. Thanks, would otherwise have forgotten about this.
